src/internal/stream/iterator.go

```diff
@@ -6,11 +6,20 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 )
 
-// EOS signals the end of the stream
-var EOS = errors.New("end of stream")
+type ErrEndOfStream struct{}
 
+func (e ErrEndOfStream) Error() string {
+	return "end of stream"
+}
+
+// EOS returns a new end of stream error
+func EOS() error {
+	return errors.EnsureStack(ErrEndOfStream{})
+}
+
+// IsEOS returns true if the error is an end of stream error.
 func IsEOS(err error) bool {
-	return errors.Is(err, EOS)
+	return errors.As(err, &ErrEndOfStream{})
```

Comment:
`errors.Is(err, ErrEndOfStream{})` should work here.  See: https://go.dev/play/p/Lpg2OyM35II